docs/pages/setup/guides/terraform-provider/reference.mdx

```diff
@@ -0,0 +1,422 @@
+Example:
+
+```
+provider "teleport" {
+  addr         = "localhost:3025"
+  cert_path    = "tf.crt"
+  key_path     = "tf.key"
+  root_ca_path = "tf.ca"
+}
+```
+
+# <a name="teleport_user"></a>`teleport_user`
+
+User resource allows to manage Teleport users: create, update, delete, assign and revoke roles and permissions.
+
+## `metadata`
+
+`metadata` is a user metadata
+
+| Name                    | Type       | Required | Description                               |
+|-------------------------|------------|----------|-------------------------------------------|
+|                   `name`|     string |        * | User name                                 |
+|            `description`|     string |          | User description                          |
+|                 `labels`| string map |          | Resource labels                           |
+|              `namespace`|     string |          | Resource namespace ("default" by default) |
+|                `expires`|     string |          | Expiry time in RFC8829 format             |
+
+Example:
+
+```
+resource "teleport_user" "example" {
+  metadata {
+    name        = "example"
+    description = "Example Teleport User"
+    expires     = "2022-10-12T07:20:50.3Z"
+    labels      = {
+      example = "yes"
+    }
+  }
+}
+```
+
+## `spec`
+
+Spec is a user spec
+
+| Name                    | Type        | Description                                                     |
+|-------------------------|-------------|-----------------------------------------------------------------|
+|                  `roles`| string list | User role names                                                 |
+|                 `traits`|         set | User traits                                                     |
+|             `traits.key`|      string | Trait name                                                      |
+|           `traits.value`| string list | Trait values                                                    |
+|      `github_identities`| object list | List of associated Github OAuth2 identities                     |
+| `github_identities.connector_id` | string | An id of registered Github connector, e.g. 'github.com'     |
+| `github_identities.username` | string | Github username                                                 |
+|      `oidc_identities`| object list | List of OpenID Connect identities                                 |
+| `oidc_identities.connector_id` | string | An id of registered OIDC connector, e.g. 'google-example.com' |
+| `oidc_identities.username` | string | OIDC username                                                     |
+|      `saml_identities`| object list | List of SAML identities                                           |
+| `saml_identities.connector_id` | string | An id of registered SAML connector, e.g. 'google-example.com' |
+| `saml_identities.username` | string | SAML username                                                     |
+
+Example:
+
+```
```

Comment:
Do you mind adding the language: `YAML` - I'm not sure this presently has any functional differences but we've planned some cool things in the future that might use these more